src/controllers/aer_controller.hpp

```diff
@@ -1524,65 +1832,161 @@ void Controller::run_circuit_without_sampled_noise(Circuit &circ,
     }
   }
 
-  // allocate qubit register
-  state.allocate(max_qubits_, block_bits);
+  // Check if measure sampling supported
+  can_sample &= check_measure_sampling_opt(circ, method);
 
-    // Check if measure sampler and optimization are valid
+  // Check if measure sampler and optimization are valid
   if (can_sample) {
     // Implement measure sampler
     auto& ops = circ.ops;
     auto first_meas = circ.first_measure_pos; // Position of first measurement op
     bool final_ops = (first_meas == ops.size());
 
+    state.set_max_matrix_bits(get_max_matrix_bits(circ) );
+
+    // allocate qubit register
+    state.allocate(circ.num_qubits, block_bits);
+
     // Run circuit instructions before first measure
     state.initialize_qreg(circ.num_qubits);
     state.initialize_creg(circ.num_memory, circ.num_registers);
+
+    RngEngine rng;
+    rng.set_seed(rng_seed);
     state.apply_ops(ops.cbegin(), ops.cbegin() + first_meas, result, rng, final_ops);
 
     // Get measurement operations and set of measured qubits
     measure_sampler(circ.ops.begin() + first_meas, circ.ops.end(), shots, state, result, rng);
 
     // Add measure sampling metadata
     result.metadata.add(true, "measure_sampling");
-  } else {
+  }
+  else{
     // Perform standard execution if we cannot apply the
     // measurement sampling optimization
-    while (shots-- > 0) {
-      run_single_shot(circ, state, result, rng);
+
+    if(batched_shots_optimization_ && batched_shots_optimization_threshold_ >= max_qubits_ 
+          && sim_device_ == Device::GPU && !cache_blocking && shots > 1 && max_batched_states_ >= num_gpus_){
```

Comment:
We can simplify this test expression by making `batch_enable` class field.

Reply:
moved to `set_parallelization_circuit()`